collections/src/main/scala/strawman/collection/mutable/ArrayBuilder.scala

```diff
@@ -69,12 +96,13 @@ object ArrayBuilder {
    *  @tparam T     type of elements for the array builder, subtype of `AnyRef` with a `ClassTag` context bound.
    */
   @SerialVersionUID(3L)
-  final class ofRef[T <: AnyRef : ClassTag] extends ArrayBuilder[T] {
+  final class ofRef[T <: AnyRef](implicit ct: ClassTag[T]) extends ArrayBuilder[T] {
 
-    private var elems: Array[T] = _
+    protected var elems: Array[T] = _
 
     private def mkArray(size: Int): Array[T] = {
-      val newelems = new Array[T](size)
+      // Anything short of this will make Dotty create an Array[Object]:
+      val newelems = java.lang.reflect.Array.newInstance(ct.runtimeClass, size).asInstanceOf[Array[T]]
```

Comment:
This is a strange one. I was unable to come up with a reproduction in the REPL but it fails consistently here. The `ClassTag` is the one for `Tuple2` but I end up with an `Array[Object]` when I use `new Array` or `Array.ofDim` (even passing the `ClassTag` explicitly).

Reply:
I opened lampepfl/dotty#4037